pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,623 @@
+	managerOs map[string]string
+	// Store the bundle load
+	bundle *manifests.Bundle
+	// containerTool defines the container tool which will be used to inspect the allOtherImages
+	containerTool string
+	// warns stores the errors faced by the validator to return the warnings
+	warns []error
+	// warns stores the errors faced by the validator to return the warnings
+	errors []error
+}
+
+// manifestInspect store the data obtained by running container-tool manifest inspect <IMAGE>
+type manifestInspect struct {
+	ManifestData []manifestData `json:"manifests"`
+}
+
+// manifestData store the platforms
+type manifestData struct {
+	Platform platform `json:"platform"`
+}
+
+// platform store the Architecture and OS supported by the image
+type platform struct {
+	Architecture string `json:"architecture"`
+	OS           string `json:"os"`
+}
+
+func multipleArchitecturesValidate(objs ...interface{}) (results []errors.ManifestResult) {
+	// Obtain the k8s version if informed via the objects an optional
+	var containerTool = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			// Check the key values informed
+			containerTool = obj.(map[string]string)[ContainerToolsKey]
+			if len(containerTool) > 0 {
+				// Make lower for we compare and use it
+				containerTool = strings.ToLower(containerTool)
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateMultiArchWith(v, containerTool))
+		}
+	}
+	return results
+}
+
+func validateMultiArchWith(bundle *manifests.Bundle, containerTool string) errors.ManifestResult {
+	result := errors.ManifestResult{}
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("bundle is nil", nil))
+		return result
+	}
+
+	result.Name = bundle.Name
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	// Validate inputs. If a container-tool key be informed
+	// with an invalid/unsupported value then make no sense do the check
+	containerTool, err := validateContainerTool(containerTool)
+	if err != nil {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+		return result
+	}
+
+	// Performs the checks
+
+	multiArchValidator := multiArchValidator{bundle: bundle, containerTool: containerTool}
+	multiArchValidator.validate()
+
+	for _, err := range multiArchValidator.warns {
+		// add the warn to the result
+		result.Add(errors.WarnFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+
+	for _, err := range multiArchValidator.errors {
+		// add the warn to the result
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+// validateContainerTool verifies if the container tool informed is valid
+func validateContainerTool(containerTool string) (string, error) {
+	if len(containerTool) == 0 || containerTool == "none" {
```

Comment:
But isn't "none" explicitly saying: "there is no container tool for this"? If that's the case, it should not ignore that setting and default to Docker.

Reply:
This is behaviour across the tools. (Keeping the same) None is == "empty"
Could you please check the commands in OPM and SDK that allows us to pass the container tools (i.e.: https://github.com/operator-framework/operator-sdk/blob/5f6c16c792b486c5d955a36b060112be288443ff/website/content/en/docs/olm-integration/cli-overview.md#private-bundle-and-catalog-image-registries)

Reply:
ah okay